src/NaadRepositoryClient.php

```diff
@@ -31,54 +39,65 @@ public function __construct()
     }
 
     /**
-     * Per NAAD documentation, certain characters must be replaced when
-     * building the URL for the short-term alert repository.
+     * Fetches an alert from the NAAD alert repository.
      *
-     * @param string $s The string to perform the replacement on.
+     * @param array $reference Heartbeat references array parts (sender, id, sent).
      *
-     * @return string
+     * @return string|null The alert response body or null if an error occurs.
+     *
+     * @throws Exception if an error occurs during the GET request.
      */
-    protected function replaceUrlCharacters( string $s ): string
+    public function fetchAlert(array $reference): ?string
```

Comment:
```suggestion
    public function fetchAlert(string $reference): ?string
```
It's still a string at this point isn't it?